File: SignalR Demo/Messaging/Blocks.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.Web;
using System.Web.Helpers;
using Microsoft.AspNet.SignalR;

namespace SignalR_Demo.Messaging
{
    public class BlocksConnection : PersistentConnection
    {
        protected override Task OnReceived(IRequest request, string connectionId, string data)
        {
            var values = Json.Decode(data);
            if (values.method == "drag")
            {
                return Connection.Broadcast(new {
                    method = "move",
                    values.top,
                    values.left,
                    values.sender
                });
            }
            
            throw new NotImplementedException();
        }
    }
    public class BlocksHub : Hub
    {
        public void Drag(Guid sender, int top, int left)
        {
            Clients.All.move(top, left, sender);
        }
    }
}

File: SignalR Demo/Scripts/apps/blocks/blocks.js
(function() {
  (function() {
    "use strict";
    var client, connection, draggable, id;
    id = guid();
    connection = $.connection('/blocksConnection');
    connection.start();
    draggable = $('.persistent [draggable]').draggable({
      drag: function(event, ui) {
        return connection.start().done(function() {
          return connection.send({
            method: "drag",
            sender: id,
            top: ui.position.top,
            left: ui.position.left
          });
        });
      }
    }).css("background-color", "#" + id.substr(0, 6));
    connection.received(function(data) {
      return client[data.method](data.top, data.left, data.sender);
    });
    return client = {
      move: function(top, left, sender) {
        if (sender !== id) {
          return draggable.css({
            top: top,
            left: left
          });
        }
      }
    };
  })();

}).call(this);
